x-pack/plugins/security_solution/public/detections/components/osquery/osquery_flyout_footer.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface EventDetailsFooterProps {
+  handleClick: () => void;
+}
+
+export const OsqueryEventDetailsFooterComponent = ({ handleClick }: EventDetailsFooterProps) => {
+  const renderFooterBody = useMemo(() => {
```

Comment:
Why have this behind a renderFn rather than putting it in the body of the component below? I believe it would work the same. It'll still re-render if the `handleClick` being passed in isn't in a `useCallback` even with the `useMemo` because that reference will change on every re-render, but if it is in a useCallback, then the useMemo isn't necessary. The React.memo around the component would be sufficient for diffing the prop at that point.

Reply:
Moved to the component below, thanks!